tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,375 @@
+        5. Update all packages, install leapp tool and fix inhibitors
+        6. Run Leapp Preupgrade and Leapp Upgrade job template
+
+    :expectedresults:
+        1. Update RHEL OS major version to another major version
+
+    """
+    logger.info('Running test for upgrade path %s', upgrade_path)
+    old_ver = custom_leapp_host.os_version.major
+    # 6. Run LEAPP-PREUPGRADE Job Template-
+    template_id = (
+        module_target_sat.api.JobTemplate()
+        .search(query={'search': 'name="Run preupgrade via Leapp"'})[0]
+        .id
+    )
+    job = module_target_sat.api.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'targeting_type': 'static_query',
+            'search_query': f'name = {custom_leapp_host.hostname}',
+        },
+    )
+    module_target_sat.wait_for_tasks(
+        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1800
+    )
+    result = module_target_sat.api.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
+    # In case of Upgrade Path A - Rehl-7 to Rhel-8
+    # Inhibitors should fix after running LEAPP-PREUPGRADE Job Template
+    if upgrade_path['source_version'].split('.')[0] == '7':
+        logger.info('Fixing inhibitory for upgrade path : %s', upgrade_path)
+        # 1. Leapp detected loaded kernel drivers which have been removed in RHEL 8
+        custom_leapp_host.run('rmmod floppy')
+        custom_leapp_host.run('rmmod pata_acpi')
+        # 2. Missing required answers in the answer file
+        leapp_report_path = "/var/log/leapp/leapp-report.txt"
+        grep_cmd = f"grep 'Title: Missing required answers in the answer file' {leapp_report_path}"
+        if custom_leapp_host.run(grep_cmd).status == 0:
+            custom_leapp_host.run(
+                'leapp answer --section remove_pam_pkcs11_module_check.confirm=True'
+            )
+
+    # Run LEAPP-UPGRADE Job Template-
+    template_id = (
+        module_target_sat.api.JobTemplate()
+        .search(query={'search': 'name="Run upgrade via Leapp"'})[0]
+        .id
+    )
+    job = module_target_sat.api.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'targeting_type': 'static_query',
+            'search_query': f'name = {custom_leapp_host.hostname}',
+            'inputs': {'Reboot': 'true'},
+        },
+    )
+    module_target_sat.wait_for_tasks(
+        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1800
+    )
+    result = module_target_sat.api.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
+    # Wait for the host to be rebooted and SSH daemon to be started.
+    custom_leapp_host.wait_for_connection()
+
+    custom_leapp_host.clean_cached_properties()
+    new_ver = custom_leapp_host.os_version.major
+    assert new_ver == old_ver + 1
```

Comment:
I think instead of old or new version check, it'd make more sense to check host version is same as target_version, right?
```suggestion
    assert custom_leapp_host.os_version == target_version
```